### ISSUE TITLE
💡 [REQUEST] - Refactoring the server/ dir to MVC model

### ISSUE THREAD

**Issue by BiswajitSahoo-tech** [anomic30/Storz#50]
### Start Date

09/04/2022

### Summary

**Below are some of my proposed changes**

**Controller**
------------------
1.  Create a Controller folder, which contains controller files
2.  There is a controller file for each route
      e,g - 
              route/user.js  <--  controller/userController.js
              route/upload.js <-- controller/uploadController.js
3. Each controller file contains the route handler function specific to its corresponding route endpoint

**Misc**
------------
1.  Merge the auth.js route to the user.js route, keeping all the authMiddleware still working. Reason for that these two route share the same API endpoint,` /api/user/`
      e,g - auth  -->  ` /api/user/login`
              user   -->  `/api/user/files`

### Basic Example

File Structure after refactoring into MVC would look like this  - 

```
controller/
    userController.js
    uploadController.js
    downloadController.js
    
model/
   
routes/
    user.js
    download.js
    upload.js
    test.js

index.js
package.json
```

### Drawbacks

Doing Refactoring to MVC only increases the require() lines and increases the number of files. Except that there is no other problem

**Comment by BiswajitSahoo-tech** [comment]
@anomic30 
Referring to PR [#18 ] data-repository/ contains a file named user.js
user.js file exports a User class
and all the method functions of this class are wrapper functions to mongoose functions
e,g - 
       instead of calling ` UserModel.findOne(fields).sort(sort);`
       now we call `User.findOne(field, sort )`  with two object parameters, which calls the above code inside of it

This kind of wrapping gives better understandability, but one drawback is that we practically can't wrap all of our mongoose code.

Thanks

**Comment by anomic30** [comment]
Yup, that's true. We should then remove this wrapper for the time being and follow the folder structure that you have mentioned. You can co-author that pull request if you want.

Thanks @BiswajitSahoo-tech

**Comment by BiswajitSahoo-tech** [comment]
Ok then, we will keep the folder structure as my suggestion
But, how to co-author a PR @anomic30

**Comment by Onesco** [comment]
@BiswajitSahoo-tech nice observation, we can if we actually want to.
